karaf/s2i/run

```diff
@@ -3,6 +3,19 @@
 # Global S2I variable setup
 source `dirname "$0"`/s2i-setup
 
+# Setup Jolokia to accept basic auth, using a randomly generated password that is stored
+# in the container in the ${DEPLOYMENTS_DIR}/jolokia.pw file.
+AB_JOLOKIA_USER="jolokia"
+if [ -f "${DEPLOYMENTS_DIR}/jolokia.pw" ] ; then
+	AB_JOLOKIA_PASSWORD=`cat "${DEPLOYMENTS_DIR}/jolokia.pw"`
+else
+	AB_JOLOKIA_PASSWORD=`tr -cd '[:alnum:]' < /dev/urandom | fold -w30 | head -n1`
+	touch "${DEPLOYMENTS_DIR}/jolokia.pw"
+	chmod 600 "${DEPLOYMENTS_DIR}/jolokia.pw"
+	echo $AB_JOLOKIA_PASSWORD > "${DEPLOYMENTS_DIR}/jolokia.pw"
+fi
+export AB_JOLOKIA_USER
```

Comment:
@chirino Just curious: Do you already had success to pass the client certification barrier when accessing via password ? I.e. how do you get hand on a signed client certificate (with private key) for your pod ? 

I might be wrong, but imo you would need to create a client certificate, let it sign by the API-server (with its CA private key, the pod obviously only has access to the public key), and then use this client certificate when accessing Jolokia. If you dont have a client cert you never would reach the authenticator because the SSL connection will reject you before this.

The only solution for this I can see is to do not the SSL client cert verification (that it is signed by the CA, remember our cert check is only an add-on checking additional features of the cert) not by the sun.HttpServer but on our own in an authenticator (and leave the SSL connector unauthenticated). Not a lightweight task.